### ISSUE TITLE
Include CompositionProvider in Starcounter

### ISSUE THREAD

**Issue by warpech** [Starcounter/Starcounter#4019]
As part of https://github.com/StarcounterPrefabs/Launcher/pull/259, a new app was extracted from Launcher: CompositionProvider. This app is really tiny - [single file](https://github.com/StarcounterPrefabs/Launcher/blob/16ea4919105cb773128dd87710d2b5d660ea699e/src/LayoutProvider/Program.cs).

This app provides HTML composition information that is used in app blending. It was one our recent goal to make this feature independent from Launcher. The need for that is discussed here: https://github.com/Starcounter/RebelsLounge/issues/47

Here's a question: How can we include CompositionProvider in Starcounter? The options are few:

1. Include this [single file](https://github.com/StarcounterPrefabs/Launcher/blob/16ea4919105cb773128dd87710d2b5d660ea699e/src/LayoutProvider/Program.cs) in Starcounter directly.
2. Keep this as a separate app that is preinstalled with Starcounter and starts automatically (still I would like to be able to upgrade it from Warehouse).
3. Keep this as a separate app that is NOT preinstalled with Starcounter. You must install it from Warehouse.

cc @alemoi, @dan31

**Comment by alemoi** [comment]
I think it should be a part of core "system" app that is started for every database.

**Comment by warpech** [comment]
How do you suggest to proceed? Move CompositionProvider app to a separate repo in Starcounter org?

**Comment by warpech** [comment]
PS. This should go in Starcounter 2.3

**Comment by tomalec** [comment]
@alemoi Could you include https://github.com/Starcounter/CompositionProvider in Starcounter? (https://github.com/StarcounterPrefabs/Launcher/pull/259#issuecomment-269080600)

**Comment by warpech** [comment]
Today it was discovered by @jwillmer that the instructions for blending given at https://docs.starcounter.io/guides/puppet-web-apps/html-view-guidelines/ **do not work** unless you have the CompositionProvider app running.

We need to:

1. Temporarily add instructions how to install CompositionProvider to the "HTML Views Guidelines" page (cc @Mackiovello)
2. Add CompositionProvider as a system app that runs automatically (see previous comments in this thread)

This is a massively strategic thing for the blending track.

cc @dan31

**Comment by tomalec** [comment]
Wow, really? Why is it so, `declarative-shadow-dom` does not work without CompositionProvider ?

**Comment by jwillmer** [comment]
No, it will add first the layout and append the slots at the bottom.

**Comment by tomalec** [comment]
Then it sounds like a bug in `starcounte-include`. Could you share the way to reproduce?
Are you sure you have Shadow DOM (polyfill) running in the browser?

**Comment by warpech** [comment]
@tomalec is right, actually `declarative-shadow-dom` should work, only the custom saved HTML Compositions do not work without the db.

I will contact @jwillmer directly to reproduce the problem, because it might be off-topic for this issue.

**Comment by Mackiovello** [comment]
> Temporarily add instructions how to install CompositionProvider to the "HTML Views Guidelines" page (cc @Mackiovello)

This is fixed for RC and Develop in https://github.com/Starcounter/Docs/commit/be3f93ae12c51291dd5fb66e830d094f74f24c24

I guess it's not applicable to the other branches since CompositionProvider do not work with them. Is Launcher a requirement on the other branches?

**Comment by tomalec** [comment]
Well in general the thing is we already published docs for V1 compatibility which is not yet ready.

Until further announcement, at our blog, you should still use `<content>`.
(We plan to release V1 compatibility by the end of this week)
